### PR TITLE
Run CI tests directly in GitHub Actions

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  # Sanity tests on Windows and MacOS.
+  otheros:
+    name: Test ${{ matrix.script }} on JDK ${{ matrix.java_version }} on ${{ matrix.os }}
+    runs-on: "${{ matrix.os }}"
+    needs: sanity
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        os: ['windows-latest', 'macos-latest']
```
Comment (Ao-senXiong) [inline .github/workflows/ci.yml:112]: I remember for mac os, we need to install timeout by `brew install coreutils`. That's why current CI are failing.

Comment (wmdietl) [reply .github/workflows/ci.yml:125]: Thanks! I think this is working now.

**Comment by cpovirk** [comment]
Thanks! I didn't review in great detail, but I did notice that we might be able to remove all the `isJava*` definitions except for `isJava8`. I did that in [a new commit on my PR](https://github.com/eisop/checker-framework/pull/823/commits/820019d366c386931b13fef9d484f21ee99bea98#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7L42)  (which also updates my PR to use the forthcoming `useJdkCompiler`), but it probably would fit better in this PR.

**Comment by Ao-senXiong** [comment]
@wmdietl Is there any specific reason we leave the misc and daikon jobs on azure?

**Pull request merged**